lua/terrortown/entities/roles/serialkiller/shared.lua

```diff
@@ -45,18 +45,30 @@ function ROLE:Initialize()
 end
 
 if SERVER then
+	--CONSTANTS
+	-- Enum for tracker mode
+	local TRACKER_MODE = {NONE = 0, RADAR = 1, TRACKER = 2}
+
 	-- Give Loadout on respawn and rolechange
 	function ROLE:GiveRoleLoadout(ply, isRoleChange)
 		ply:GiveEquipmentWeapon("weapon_ttt_sk_knife")
-		ply:GiveEquipmentItem("item_ttt_tracker")
-		ply:GiveArmor(60)
+		if GetConVar("ttt2_serialkiller_tracker_mode"):GetInt() == TRACKER_MODE.RADAR then
+			ply:GiveEquipmentItem("item_ttt_radar")
+		elseif GetConVar("ttt2_serialkiller_tracker_mode"):GetInt() == TRACKER_MODE.TRACKER then
+			ply:GiveEquipmentItem("item_ttt_tracker")
+		end
+		ply:GiveArmor(GetConVar("ttt2_serialkiller_armor"):GetInt())
 	end
 
 	-- Remove Loadout on death and rolechange
 	function ROLE:RemoveRoleLoadout(ply, isRoleChange)
 		ply:StripWeapon("weapon_ttt_sk_knife")
-		ply:RemoveEquipmentItem("item_ttt_tracker")
-		ply:RemoveArmor(60)
+		if GetConVar("ttt2_serialkiller_tracker_mode"):GetInt() == TRACKER_MODE.RADAR then
+			ply:RemoveEquipmentItem("item_ttt_radar")
+		elseif GetConVar("ttt2_serialkiller_tracker_mode"):GetInt() == TRACKER_MODE.TRACKER then
+			ply:RemoveEquipmentItem("item_ttt_tracker")
```

Comment:
That does not do the same as the code below, because the tracker item doesn't highlight Jesters (as the removed code will do)

Reply:
Yes, I did briefly test out that old code before removing it. From what I can tell the tracker item overwrites that code block, which highlights jesters as pink and everyone else as green. I have never seen this occur in game (prior to modding), and assumed the code block was overwritten on purpose. I can't tell if you want the old code to still be available to players through some fashion or not from your comment. If you do, it can always be added to TRACKER_MODE as a fourth option.

Reply:
Hmmm.. Then it's probably something for the tracker item to implement and I'm okay with it.